### DISCUSSION TITLE
Conceptual issue with date used for sorting chat messages

### DISCUSSION THREAD

**Discussion by HenrikJannsen** [bisq-network/bisq2#1274] [General]
Currently we use the chat message's creation date for sorting in the chat message list view.
With I2P/Tor it can be that a message arrives with quite a delay and then might get inserted into previous messages which have been received faster. This is confusing as the most recently received message would not be the last message in the list.

If we would use the date when we received the message (e.g. from the data storage layer where we have the date for checking if a message is expired) we would get all messages with the same date at first time opening the app or when we have not been online and get a bunch of messages. So the sorting would also messed up. Beside that the current API does not expose that receive date to client code.

Seems there is no good solution and keeping it as it is might be the lesser evil.

**Comment by salokiam** [comment]
How much of a time lag are we talking about?

When I start up Bisq2, I would want it to be sorted by sendate.

After Startup, when it catches up and download all messages since I was last online, I would like it to be sorted by sendDate as well.

Only after I read the latest message, and then a mesage arrives, which is actually older than the last message, it would be inserted somewhere in the messagestream. That could be strange. An indicator that shows, that this individual message is new would probably be enough. But even without an indicator, sorting it by sendDate should be fine for me, as I consider this as an edge case.

**Reply by HenrikJannsen** [reply]
Agree, we can ignore it as edge case for now and use some indicator later to cover those. Will close the discussion.
